WcaOnRails/lib/qualification.rb

```diff
@@ -35,11 +35,36 @@ def self.load(json)
       rescue ArgumentError
         nil
       end
-      p out
       out
     end
   end
 
+  def can_register?(user, event_id)
```

Comment:
Debating with myself whether this should be split up into `qualifying_results_for(user, event_id)` here and the actual `can_register?` over in `competition_event`. I have no strong arguments other than gut feeling. What is your opinion?

Reply:
The tricky thing here is "ranking" qualifications. If we did that, the code in `qualification.rb` would either need to ignore "ranking", or be more complicated to consider registered competitors (and then have `CompetitionEvent.can_register?` ignore it). I went with this naming scheme to accurately represent what the code actually checks: whether, based on the qualification standard, the competitor is allowed to register for the event.

For what it's worth, the code to remove unqualified competitors won't use this; it'll do one big SQL query to generate the psych sheet as of the deadline, to avoid doing N SQL queries for N competitors.